core/src/tile/tileWorker.cpp

```diff
@@ -144,6 +185,10 @@ void TileWorker::setScene(std::shared_ptr<Scene>& _scene) {
     }
 }
 
+void TileWorker::notifyAll() {
+    m_condition.notify_all();
+}
+
```

Comment:
It will block on ```wait``` *once the condition is not met*, but since the re-enqueue causes either another worker thread to wake up (with notify_one) or lets the current worker continue to run in the next ```while(true)``` iteration since ```!m_queue.empty()```